api/src/services/setup/upgrade-steps.js

```diff
@@ -116,7 +116,20 @@ const complete = async (buildInfo) => {
   const stagingDoc = await upgradeUtils.getStagingDoc(buildInfo);
   const payload = upgradeUtils.getUpgradeServicePayload(stagingDoc);
 
-  return await upgradeUtils.makeUpgradeRequest(payload);
+  const response = await upgradeUtils.makeUpgradeRequest(payload);
+
+  if (!upgradeUtils.upgradeResponseSuccess(payload, response)) {
```

Comment:
From a design perspective it would be better to do this work in the `makeUpgradeRequest` function rather than having to make a second call to work out if the first call was successful or not.